### PR TITLE
Fix "Bump changelog on release" GH Action

### LINKED ISSUE #1185: Fix bump changelog action
https://github.com/pypa/pipx/pull/1184#issue-2062575646

### PR DESCRIPTION
**Pull request by chrysle**
- [ ] I have added an entry to `docs/changelog.md`

## Summary of changes

Basically I made sure I'm allowed to bypass the branch protection rules and configured the action to commit with my identity.
I think we shouldn't remove the branch protection.

This ought to close #1185.

## Test plan

Tested by running

```
# command(s) to exercise these changes
```

### CODE REVIEW

**Comment by henryiii** [comment]
Wouldn't it be better to have it make a PR? That's what I was intending to do later.

**Comment by chrysle** [comment]
@henryiii I thought merging that every time would mean increased effort, but I guess you're right. If you don't insist on doing it, I could also configure it in this PR (I'd use GH CLI).

**Comment by henryiii** [comment]
You are welcome to do it. :) I think you can use the gh CLI, I've intended to try with that, I've used https://github.com/peter-evans/create-pull-request before and it works well. You do have to close and reopen to trigger CI unless you set up a bot, which I haven't done yet.

https://github.com/scientific-python/cookie/blob/main/.github/workflows/bump.yml is one I've set up, another is https://github.com/pypa/cibuildwheel/blob/main/.github/workflows/update-dependencies.yml which does have the bot.

**Comment by chrysle** [comment]
I've already pushed an attempt, could you have a look?

**Comment by henryiii** [comment]
I haven't tried gh for this, so I don't want to guarantee anything, but it looks correct and like what I would have tried, I think.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -135,11 +135,19 @@ jobs:
...
       - name: Extract release tag
         run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Create pull request branch
+        run: git switch -c "bump-changelog-for-${RELEASE_VERSION}"
       - name: Update changelog
         run: echo -e "## dev\n\n## $RELEASE_VERSION\n$(tail -n +2 CHANGELOG.md)" > CHANGELOG.md
       - name: Commit and push change
         run: |
-          git config --global user.name 'Github Actions'
+          git config --global user.name 'GitHub Actions'
```
Comment (dukecat0) [inline .github/workflows/tests.yml:144]:
```suggestion
          git config --global user.name "github-actions[bot]"
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -135,11 +135,19 @@ jobs:
...
           git config --global user.email 'gh-action@users.noreply.github.com'
```
Comment (dukecat0) [inline .github/workflows/tests.yml:145]:
```suggestion
          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

**Comment by chrysle** [comment]
@henryiii @dukecat0 I don't seem to be able to test this fully on my fork (see this [actions run](https://github.com/chrysle/pipx/actions/runs/7433631407/job/20226961823)), but judging by what _does_ work there, this should be ready.

**Comment by henryiii** [comment]
Do you need the `pull-requests: write` permission? I noticed I have that one one that uses `peter-evans/create-pull-request@v5`. (I'm not sure how to nicely do a "Make a pull request only if changed", or I would have projects to test out `gh pr` in CI).

https://github.com/scientific-python/cookie/blob/57fc35295e5cfc79b9152a82d183cb8c73c6865b/.github/workflows/bump.yml#L9

**Comment by gaborbernat** [comment]
@chrysle this failed https://github.com/pypa/pipx/actions/runs/7503796146/job/20430009811 ...

**Comment by henryiii** [comment]
Fix in #1202, it was missing `pull-requests: write`.

**Pull request merged**